lib/media/segment_index.js

```diff
@@ -700,6 +700,25 @@ shaka.media.MetaSegmentIndex = class extends shaka.media.SegmentIndex {
     this.indexes_ = [];
   }
 
+  /** Evicts all old SegmentIndexes in this MetaSegmentIndex that are empty. */
+  evictEmpty() {
+    // If we hypothetically have a segment index that is empty but comes after
+    // one that has references, assume it's new and leave it in.
+    const newIndexes = [];
+    let include = false;
+    for (const index of this.indexes_) {
+      if (!index.isEmpty()) {
+        include = true;
+      }
+      if (include) {
+        newIndexes.push(index);
+      } else {
+        index.release();
+      }
+    }
+    this.indexes_ = newIndexes;
+  }
```

Comment:
1. Can we just evict as long as we find first non empty segment index?
2. We should double ensure we always keep at least the newest segment index, even if it's empty.

Reply:
1. Is this what you meant?
2. Added an assert.